beancount_import/source/amazon_invoice.py

```diff
@@ -15,6 +17,256 @@
+    digital_payment_information: str
+
+    @staticmethod
+    @abstractmethod
+    def parse_amount(amount, assumed_currency=None) -> Amount:
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def parse_date(date_str) -> datetime.date:
+        raise NotImplementedError
+
+
+class Locale_en_US(Locale_Data):
+    """Language and region specific settings for parsing amazon.com invoices
+    """
+    LOCALE='en_US'
+    tax_included_in_price=False
+    payee='Amazon.com'
+    
+    # common fields regular and digital orders
+    items_ordered='Items Ordered' # shipment + digital
+    price='Price' # shipment + digital
+    currency='USD' # shipment only
+    items_subtotal=r'Item\(s\) Subtotal:' # shipment +digital
+    total_before_tax='Total Before Tax:' # shipment + digital
+    pretax_adjustment_fields_pattern=('(?:' + '|'.join([
+        'Shipping & Handling', # Verpackung & Versand:
+        'Free Shipping',
+        'Free delivery',
+        'Pantry delivery',
+        'Promotion(?:s| Applied)', # Gutschein eingelöst:
+        'Lightning Deal',
+        'Your Coupon Savings', 
+        '[0-9]+% off savings',
+        'Subscribe & Save',
+        '[0-9]+ Audible Credit Applied',
+        '.*[0-9]+% Off.*',
+        'Courtesy Credit',
+        'Extra Savings',
+        '(?:.*) Discount',
+        'Gift[ -]Wrap',
+    ]) + ') *:')
+    posttax_adjustment_fields_pattern=r'Gift Card Amount:|Rewards Points:|Tip [(]optional[)]:|Recycle Fee \$X'
+
+    # Payment Table & Credit Card Transactions
+    grand_total=r'\n\s*Grand Total:\s+(.*)\n'
+    credit_card_transactions='Credit Card transactions'
+    credit_card_last_digits=r'^([^:]+) ending in ([0-9]+):\s+([^:]+):$'
+    payment_type=[
+        # only first matching regex is used!
+        r'\n\s*([^\s|][^|\n]*[^|\s])\s+\|\s+Last (?:4 )?digits:\s+([0-9]{4})\n',
+        r'\n\s*(.+)\s+ending in\s+([0-9]{4})\n'
+        ]
+    payment_information='^Payment information$'
+
+    # regular orders only
+    shipment_shipped_pattern='^Shipped on ([^\\n]+)$'
+    shipment_nonshipped_headers=[
+        'Service completed',
+        'Preparing for Shipment',
+        'Not Yet Shipped',
+        'Shipping now'
+        ]
+    shipment_quantity=r'^\s*(?:(?P<quantity>[0-9]+)|(?P<weight1>[0-9.]+\s+(?:lb|kg))|(?:(?P<quantityIgnore>[0-9.]+) [(](?P<weight2>[^)]+)[)]))\s+of:'
+    shipment_of='of:'
+    shipment_sales_tax='Sales Tax:'
+    shipment_total='Total for This Shipment:'
+    shipment_seller_profile=' (seller profile)'
+    shipment_sold_by=r'(?P<description>.*)\n\s*(?:Sold|Provided) by:? (?P<sold_by>[^\n]+)'
+    shipment_condition=r'\n.*\n\s*Condition: (?P<condition>[^\n]+)'
+    regular_total_order='Grand Total:'
+    regular_estimated_tax = 'Estimated tax to be collected:'
+    regular_order_placed=r'(?:Subscribe and Save )?Order Placed:\s+([^\s]+ \d+, \d{4})'
+    regular_order_id=r'.*Order ([0-9\-]+)'
+    gift_card='Gift Cards' # not confirmed yet!
```

Comment:
I don't know how gift cards are handled on `.com`, these are just guesses inferred from `.de`. There (electronic) giftcards are handled differently than regular shipments. They have an own "shipment table".

Everything is handled in the new `parse_gift_cards` method.
If there is no table node with text matching`gift_card`, the code doesn't do anything. (probably the case for `en_US`) I decided not to add an additional boolean flag to enable/disable the `parse_gift_cards` method.

Reply:
Here I also think we'd better not guess the correct string to match, but fail gracefully.

Reply:
if it is not found, it does not even fail. But I made the attribute optional now.

I also added a check if there are no items found at all for the regular order or digital order (no shipments, no gift cards), this should indicate that there is something wrong with the parsing...